pyiceberg/catalog/__init__.py

```diff
@@ -627,6 +632,25 @@ def identifier_to_tuple_without_catalog(self, identifier: Union[str, Identifier]
             identifier_tuple = identifier_tuple[1:]
         return identifier_tuple
 
+    def _identifier_to_tuple_without_catalog(self, identifier: Union[str, Identifier]) -> Identifier:
```

Comment:
where as this is now called by the PyIceberg functions and only prints the deprecation warning message if we are using the unsupported naming convention